src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -1,108 +1,918 @@
+    counterDigits[1] = 15;
+    counterDigits[0] = counterDigits[1] / 10;
+    counterDigits[1] -= (s16)(counterDigits[0] * 10);
 
-s32 D_8082B538[] = {
-    0x00000400,
-    0x00000000,
-    0x00000000,
-    0x00000000,
-};
+    // Draw max number of stray fairies
+    for (rectLeft = 116, digitIndex = 0; digitIndex < 2; digitIndex++, rectLeft += 8) {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, 255);
+
+        POLY_OPA_DISP = func_8010D7D0(POLY_OPA_DISP, (u8*)gCounterDigit0Tex + (8 * 16 * counterDigits[digitIndex]), 8,
+                                      16, rectLeft + 1, 146, 8, 16, 1 << 10, 1 << 10);
+
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+
+        gSPTextureRectangle(POLY_OPA_DISP++, rectLeft * 4, 580, (rectLeft * 4) + 32, 644, G_TX_RENDERTILE, 0, 0,
+                            1 << 10, 1 << 10);
+    }
+
+    // Draw Counter slash
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, 255);
+
+    POLY_OPA_DISP =
+        func_8010D7D0(POLY_OPA_DISP, gStrayFairyMapCounterSlashTex, 8, 16, 107, 146, 8, 16, 1 << 10, 1 << 10);
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+
+    gSPTextureRectangle(POLY_OPA_DISP++, 424, 580, 456, 644, G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
+
+    // Get digits for current number of stray fairies collected
+    counterDigits[1] = gSaveContext.save.inventory.strayFairies[(void)0, gSaveContext.dungeonIndex];
+    counterDigits[0] = counterDigits[1] / 10;
+    counterDigits[1] -= (s16)(counterDigits[0] * 10);
+
+    // Draw digits for current number of stray fairies collected
+    for (rectLeft = 88, digitIndex = 0; digitIndex < 2; digitIndex++, rectLeft += 8) {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, 255);
+
+        POLY_OPA_DISP = func_8010D7D0(POLY_OPA_DISP, (u8*)gCounterDigit0Tex + (8 * 16 * counterDigits[digitIndex]), 8,
+                                      16, rectLeft + 1, 146, 8, 16, 1 << 10, 1 << 10);
+
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+
+        gSPTextureRectangle(POLY_OPA_DISP++, rectLeft * 4, 580, (rectLeft * 4) + 32, 644, G_TX_RENDERTILE, 0, 0,
+                            1 << 10, 1 << 10);
+    }
 
-s16 D_8082B548[] = {
-    0x0C00, 0x8418, 0x0C00, 0x9098, 0x0C01, 0x2898, 0x0C01, 0x35D8, 0x0C01, 0x4318,
-    0x0C00, 0xDE58, 0x0C00, 0xEB58, 0x0C00, 0xFAA8, 0x0C00, 0x9858, 0x0C00, 0x9E58,
-    0x0C00, 0xAE58, 0x0C00, 0xBE58, 0x0C00, 0xCE58, 0x0C01, 0x09F8, 0x0C01, 0x1958,
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+TexturePtr sDungeonItemTextures[] = {
+    0x09003600, // `gBossKeyIconTex`: DUNGEON_BOSS_KEY
+    0x09003F00, // `gCompassIconTex`: DUNGEON_COMPASS
+    0x09004800, // `gDungeonMapIconTex`: DUNGEON_MAP
+};
```

Comment:
These are in `iconItem24Segment` which is an archive format file. We don't have the proper support for that yet, although @Thar0 has the scripts setup so it's not far from support I believe.

You'll notice in `z_kaleido_scope_NES` that there's also quite a few textures from `iconItemSegment`, which is also an archive format file, so it would be nice to have that setup soonish, but we'll be fine if its done later.

This archive and build script stuff is still over my head, so I'm not sure how much I could help with this